portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1899,6 +1901,20 @@ public void close() throws IOException {
 		}
 	}
 
+	private URI _getURI(String url) throws URISyntaxException {
+		Map<String, URI> uriMap = _uriMap.get();
+
+		URI uri = uriMap.get(url);
+
+		if (uri == null) {
+			uri = new URI(url);
+
+			uriMap.put(url, new URI(url));
```

Comment:
@csierra you have a double creation for the URI here, please fix it and rerun the performance test, the regression should get smaller after the fix.